luaa.c

```diff
@@ -19,14 +19,17 @@
  *
  */
 
-/** awesome core API
+/** AwesomeWM lifecycle API.
```

Comment:
The changes to `luaa.c` also look quite simple and non-controversial. Could be moved to a separate PR, if you want. (After incorporating Daniel's feedback.)